File: C++/Step14-Math_3 Number Theory and Combinatorics/2609.cpp
//
//  2609.cpp
//  Step14-Math_3 Number Theory and Combinatorics
//
//  Created by 김예빈 on 2019. 9. 15..
//  Copyright © 2019년 김예빈. All rights reserved.
//

#include <iostream>
using namespace std;

int gcd(int, int);

int main(int argc, const char * argv[]) {
    cin.tie(NULL);
    ios::sync_with_stdio(false);
    
    int a, b, greatest, least;
    
    cin >> a >> b;
    
    greatest = a > b ? gcd(a, b) : gcd(b, a);
    least = a * b / greatest;
    
    cout << greatest << "\n" << least;
    
    return 0;
}

int gcd(int a, int b) {
    if (a % b == 0)
        return b;
    else
        return gcd(b, a % b);
}

File: BaekJoon Online Judge/Swift/etc/2609.swift
//
//  2609.swift
//  Step16-Math_3 Number Theory and Combinatorics
//
//  Created by Yebin Kim on 2019/12/23.
//  Copyright © 2019 김예빈. All rights reserved.
//

import Foundation

func euclidean(_ a: Int, _ b: Int) -> Int {
    if b % a == 0 {
        return a
    } else {
        return euclidean(b % a, a)
    }
}

let input = readLine()!.split(separator: " ").map{ Int($0)! }.sorted()
let A = input[0]
let B = input[1]

let gcd = euclidean(A, B)

print(gcd)
print(A * B / gcd)
